java/test/org/openqa/selenium/environment/webserver/CookieHandler.java

```diff
@@ -116,9 +116,9 @@ private Collection<Cookie> getCookies(HttpRequest request) {
   private void addCookie(HttpResponse response, Cookie cook) {
     StringBuilder cookie = new StringBuilder();
 
-    // TODO: escape string as necessary
-    String name = cook.getName();
-    cookie.append(name).append("=").append(cook.getValue()).append("; ");
+    String name = escapeCookieValue(cook.getName());
+    String value = escapeCookieValue(cook.getValue());
```

Comment:
Both name and value need to be escaped because it can have inputs like `"user\"Name"` and `John;Doe`.